pkg/collector/corechecks/system/battery/battery_windows.go

```diff
@@ -0,0 +1,422 @@
+	if err := c.CommonConfigure(senderManager, initConfig, data, source); err != nil {
+		return err
+	}
+
+	// Check if battery is available before enabling the check
+	log.Debugf("Checking if battery is available")
+	hasBattery, err := HasBatteryAvailable()
+	if err != nil {
+		return err
+	}
+	if !hasBattery {
+		log.Infof("No battery available, skipping check")
+		return check.ErrSkipCheckInstance
+	}
+
+	return nil
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+
+	info, err := QueryBatteryInfo()
+	if err != nil {
+		return err
+	}
+
+	sender.Gauge("system.battery.designed_capacity", info.DesignedCapacity, "", nil)
+	sender.Gauge("system.battery.maximum_capacity", info.FullChargedCapacity, "", nil)
+	sender.Gauge("system.battery.maximum_capacity_pct", info.MaximumCapacityPct, "", nil)
+	sender.Gauge("system.battery.cycle_count", info.CycleCount, "", nil)
+	sender.Gauge("system.battery.current_charge", info.CurrentCharge, "", nil)
+	sender.Gauge("system.battery.voltage", info.Voltage, "", nil)
+	sender.Gauge("system.battery.charge_rate", info.ChargeRate, "", nil)
+
+	if len(info.PowerState) > 0 {
+		sender.Gauge("system.battery.power_state", 1, "", info.PowerState)
+	} else {
+		sender.Gauge("system.battery.power_state", 0, "", []string{"power_state:unknown"})
+	}
+
+	sender.Commit()
+	return nil
+}
+
+// hasBatteryAvailable checks if at least one battery device is present
+func hasBatteryAvailable() (bool, error) {
+	hdev, err := windows.SetupDiGetClassDevsEx(&GUID_DEVCLASS_BATTERY, "", 0, windows.DIGCF_PRESENT|windows.DIGCF_DEVICEINTERFACE, 0, "")
+	if err != nil {
+		return false, fmt.Errorf("SetupDiGetClassDevs: %w", err)
+	}
+	defer func() {
+		err := windows.SetupDiDestroyDeviceInfoList(hdev)
+		if err != nil {
+			log.Errorf("Error destroying device info list: %v", err)
+		}
+	}()
+
+	var ifData winutil.SP_DEVICE_INTERFACE_DATA
+	ifData.CbSize = uint32(unsafe.Sizeof(ifData))
+
+	err = winutil.SetupDiEnumDeviceInterfaces(hdev, &GUID_DEVCLASS_BATTERY, 0, &ifData)
+	if err != nil {
+		// No battery devices found
+		log.Debugf("No battery devices found")
+		return false, nil
+	}
+
+	// At least one battery device exists
+	log.Debugf("At least one battery device exists")
+	return true, nil
+}
+
+func queryBatteryInfo() (*BatteryInfo, error) {
+	info := &BatteryInfo{}
+	hdev, err := windows.SetupDiGetClassDevsEx(&GUID_DEVCLASS_BATTERY, "", 0, windows.DIGCF_PRESENT|windows.DIGCF_DEVICEINTERFACE, 0, "")
+	if err != nil {
+		return nil, fmt.Errorf("SetupDiGetClassDevs: %w", err)
+	}
+	defer func() {
+		err := windows.SetupDiDestroyDeviceInfoList(hdev)
+		if err != nil {
+			log.Errorf("Error destroying device info list: %v", err)
+		}
+	}()
+
+	var ifData winutil.SP_DEVICE_INTERFACE_DATA
+	ifData.CbSize = uint32(unsafe.Sizeof(ifData))
+
+	for i := uint32(0); ; i++ {
+		err = winutil.SetupDiEnumDeviceInterfaces(hdev, &GUID_DEVCLASS_BATTERY, i, &ifData)
```

Comment:
Are we picking first found battery? What is it is a keyboard battery, probably people are interested in that

Reply:
sounds good